### DISCUSSION TITLE
Contiki-ng sdn(software defined network) solution.

### DISCUSSION THREAD

**Discussion by aceylan** [contiki-ng/contiki-ng#1790] [Q&A]
Hi. I am nearly new on contiki-ng. 
I want to make an sdn solution on contiki and cooja. For this reason, I need to connect to a mote from my python app. I guess I may connect and communicate with a mote over serial socket connection. 
But I don't know how to do this. Maybe I can open a socket from a port like in the example of rpl-border-router via cooja and make a serial connection to it with my python script. I don't know if it's possible. If there is an example, I would appreciate it if you could tell me.
I saw a solution called it-sdn, but the source codes are not available on the internet at the moment.
Apart from that, I have a few more questions.
How can I disable rpl routing?
How can I communicate between motes at a basic level? Which would you recommend I communicate at the mac layer or the ip layer.
I would appreciate it if you could tell me the steps to follow. 
I hope I explained well. I am sorry for my English.
Thanks.

**Comment by mbaddeley** [comment]
Hi aceylan, I did a (mostly) working port of uSDN to NG a while back. You can find it here:

https://github.com/mbaddeley/usdn-ng
